src/components/notifications/NotificationsMenu.js

```diff
@@ -149,6 +133,56 @@ function NotificationsMenu(props) {
         },
     });
 
+    const [markAllSeenMutation] = useMutation({
+        queryKey: NOTIFICATIONS_MARK_ALL_SEEN_KEY,
+        queryFn: markAllSeen,
+        config: {
+            onSuccess: () => {
+                setAllNotificationsSeen();
+            },
+            onError: (error) => {
+                console.log("Error marking all notifications as Seen");
+                // *** Will add this after understanding HOC better. ***
+                // showErrorAnnouncer(
+                //     t("errorMarkAsSeen", {
+                //         count: notifications.length,
+                //     }),
+                //     error
+                // );
+            },
+        },
+    });
+
+    const setAllNotificationsSeen = () => {
+        setNotifications(
+            notifications?.map((item) => {
+                const tempItem = { ...item };
+                tempItem.seen = true;
+                return tempItem;
+            })
+        );
+        setUnSeenCount(0);
+    };
+
+    const handleClose = () => {
+        setAnchorEl();
+    };
+
+    const handleMarkAllAsSeenClick = () => {
+        markAllSeenMutation();
+        setAllNotificationsSeen();
```

Comment:
Having `setAllNotificationsSeen()` get called here when the user clicks the `Mark All As Seen` button means that this will trigger even if the API call to set all notifications as seen fails.  I don't think you need this here - having the call in `onSuccess` like you already have should be sufficient.